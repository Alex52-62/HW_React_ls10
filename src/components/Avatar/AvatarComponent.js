import React from "react";

import { AvatarsView } from "./AvatarView";
import "./style.css";

export const Ava = () => {
  const avatars = [
    { nickname: "Dreamer", name: "James" },
    { nickname: "Detonator", name: "Jorgen" },
    { nickname: "SSSR", name: "Amy" },
    { nickname: "Crazy", name: "Max" },
    { nickname: "KickBoy", name: "Phil" },
    { nickname: "Falcon", name: "Wim" },
    { nickname: "Angel", name: "John" },
  ];

  const result = avatars.map((avatar) => {
    return avatar.nickname;
  });

  return (
    <div className="avatar">
      <AvatarsView result={result} />
    </div>
  );
};
