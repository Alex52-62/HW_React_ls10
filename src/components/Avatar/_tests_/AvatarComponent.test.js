import React from "react";
import { render, mock, shallow } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Ava } from "../AvatarComponent";

describe("Avatar", () => {
  it("matches snapshot", () => {
    const component = render(<Ava />);

    expect(component).toMatchSnapshot();
  });
});
