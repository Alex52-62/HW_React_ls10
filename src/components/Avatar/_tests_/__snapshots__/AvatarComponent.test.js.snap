// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="avatar"
      >
        <div
          class="css-12g71p8"
        >
          <img
            alt="ui-avatar-Dreamer"
            class=" css-tn5ehy"
            draggable="false"
            size="60"
            src="https://ui-avatars.com/api/?size=120&name=Dreamer&font-size=0.66&color=FFFFFF&background=2196f3&uppercase=false&length=1"
          />
          <img
            alt="ui-avatar-Detonator"
            class=" css-tn5ehy"
            draggable="false"
            size="60"
            src="https://ui-avatars.com/api/?size=120&name=Detonator&font-size=0.66&color=FFFFFF&background=009688&uppercase=false&length=1"
          />
          <img
            alt="ui-avatar-SSSR"
            class=" css-tn5ehy"
            draggable="false"
            size="60"
            src="https://ui-avatars.com/api/?size=120&name=SSSR&font-size=0.66&color=FFFFFF&background=9c27b0&uppercase=false&length=1"
          />
          <img
            alt="ui-avatar-Crazy"
            class="hidden css-tn5ehy"
            draggable="false"
            size="60"
            src="https://ui-avatars.com/api/?size=120&name=Crazy&font-size=0.66&color=FFFFFF&background=ffc107&uppercase=false&length=1"
          />
          <img
            alt="ui-avatar-KickBoy"
            class="hidden css-tn5ehy"
            draggable="false"
            size="60"
            src="https://ui-avatars.com/api/?size=120&name=KickBoy&font-size=0.66&color=FFFFFF&background=e91e63&uppercase=false&length=1"
          />
          <img
            alt="ui-avatar-Falcon"
            class="hidden css-tn5ehy"
            draggable="false"
            size="60"
            src="https://ui-avatars.com/api/?size=120&name=Falcon&font-size=0.66&color=FFFFFF&background=ff9800&uppercase=false&length=1"
          />
          <img
            alt="ui-avatar-Angel"
            class="hidden css-tn5ehy"
            draggable="false"
            size="60"
            src="https://ui-avatars.com/api/?size=120&name=Angel&font-size=0.66&color=FFFFFF&background=2196f3&uppercase=false&length=1"
          />
          <img
            alt="ui-avatar-+4"
            class=" css-gx2f49"
            draggable="false"
            size="60"
            src="https://ui-avatars.com/api/?size=120&name=%2B4&font-size=0.66&color=795548&background=FFFFFF&uppercase=false&length=5"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="avatar"
    >
      <div
        class="css-12g71p8"
      >
        <img
          alt="ui-avatar-Dreamer"
          class=" css-tn5ehy"
          draggable="false"
          size="60"
          src="https://ui-avatars.com/api/?size=120&name=Dreamer&font-size=0.66&color=FFFFFF&background=2196f3&uppercase=false&length=1"
        />
        <img
          alt="ui-avatar-Detonator"
          class=" css-tn5ehy"
          draggable="false"
          size="60"
          src="https://ui-avatars.com/api/?size=120&name=Detonator&font-size=0.66&color=FFFFFF&background=009688&uppercase=false&length=1"
        />
        <img
          alt="ui-avatar-SSSR"
          class=" css-tn5ehy"
          draggable="false"
          size="60"
          src="https://ui-avatars.com/api/?size=120&name=SSSR&font-size=0.66&color=FFFFFF&background=9c27b0&uppercase=false&length=1"
        />
        <img
          alt="ui-avatar-Crazy"
          class="hidden css-tn5ehy"
          draggable="false"
          size="60"
          src="https://ui-avatars.com/api/?size=120&name=Crazy&font-size=0.66&color=FFFFFF&background=ffc107&uppercase=false&length=1"
        />
        <img
          alt="ui-avatar-KickBoy"
          class="hidden css-tn5ehy"
          draggable="false"
          size="60"
          src="https://ui-avatars.com/api/?size=120&name=KickBoy&font-size=0.66&color=FFFFFF&background=e91e63&uppercase=false&length=1"
        />
        <img
          alt="ui-avatar-Falcon"
          class="hidden css-tn5ehy"
          draggable="false"
          size="60"
          src="https://ui-avatars.com/api/?size=120&name=Falcon&font-size=0.66&color=FFFFFF&background=ff9800&uppercase=false&length=1"
        />
        <img
          alt="ui-avatar-Angel"
          class="hidden css-tn5ehy"
          draggable="false"
          size="60"
          src="https://ui-avatars.com/api/?size=120&name=Angel&font-size=0.66&color=FFFFFF&background=2196f3&uppercase=false&length=1"
        />
        <img
          alt="ui-avatar-+4"
          class=" css-gx2f49"
          draggable="false"
          size="60"
          src="https://ui-avatars.com/api/?size=120&name=%2B4&font-size=0.66&color=795548&background=FFFFFF&uppercase=false&length=5"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
